### PR TITLE
feat(cli): cli new help UX （new, new template）

### PR DESCRIPTION
**Pull request by jayzhang**
feature link: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/24539544

New CLI -help UI (for "new", "new template" command):

image

**Note: this PR will not affect current CLI behavior, it will be enabled under feature flag in the future.**

### CODE REVIEW

File: packages/cli/src/commonlib/logger.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (swatDong) [inline packages/cli/src/commonlib/logger.ts:1]: Is this a renaming of *cli/src/commonlib/**log.ts***?

Comment (jayzhang) [reply packages/cli/src/commonlib/logger.ts:1]: logger.ts is a refactored log provider which will replace log.ts in the future

Comment (swatDong) [reply packages/cli/src/commonlib/logger.ts:1]: So are you going to do the refactor and own the new logger? If so, feel free to change the CODEOWNERS then no need to wait unnecessary reviewers each time~

**Review by Alive-Fish — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]
Check if "npx teamsfx xxx" works.

**Review by swatDong — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Review by sffamily — APPROVED** [review]

**Pull request merged**